clients/src/main/java/org/apache/kafka/clients/CommonClientConfigs.java

```diff
@@ -194,8 +194,8 @@
     public static Map<String, Object> postProcessReconnectBackoffConfigs(AbstractConfig config,
                                                     Map<String, Object> parsedValues) {
         HashMap<String, Object> rval = new HashMap<>();
-        if ((!config.originals().containsKey(RECONNECT_BACKOFF_MAX_MS_CONFIG)) &&
-                config.originals().containsKey(RECONNECT_BACKOFF_MS_CONFIG)) {
+        if ((!config.hasKeyInOriginals(RECONNECT_BACKOFF_MAX_MS_CONFIG)) &&
+                config.hasKeyInOriginals(RECONNECT_BACKOFF_MS_CONFIG)) {
```

Comment:
4. refactor the `originals().containsKey` usage. In `originals()`, we'll make a copy of the configs, and in most cases, we only want to check `containsKey`. Refactor it by directly check the key in `AbstractConfigs` to avoid unnecessary map copy.